pkg/controller/podupdater/pod_updater_controller.go

```diff
@@ -0,0 +1,502 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podupdater
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	intstrutil "k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	appsinformers "k8s.io/client-go/informers/apps/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	appslisters "k8s.io/client-go/listers/apps/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	k8sutil "github.com/openyurtio/openyurt/pkg/controller/podupdater/kubernetes"
+)
+
+const (
+	// UpgradeAnnotation is the annotation key used in daemonset spec to indicate
+	// which upgrade strategy is selected. Currently "ota" and "auto" are supported.
+	UpgradeAnnotation = "apps.openyurt.io/upgrade-strategy"
+
+	OTAUpgrade  = "ota"
+	AutoUpgrade = "auto"
+)
+
+// PodUpgradableAnnotation is the annotation key added to pods to indicate
+// whether a new version is available for upgrade.
+// This annotation will only be added if the upgrade strategy is "apps.openyurt.io/upgrade-strategy":"ota".
+const PodUpgradableAnnotation = "apps.openyurt.io/pod-upgradable"
+
+const MaxUnavailableAnnotation = "apps.openyurt.io/max-unavailable"
+
+const (
+	// BurstReplicas is a rate limiter for booting pods on a lot of pods.
+	// The value of 250 is chosen b/c values that are too high can cause registry DoS issues.
+	BurstReplicas = 250
+)
```

Comment:
please merge line 48~68 into one `const`

Reply:
I separate these constant for different usage before. I'll merge them.